src/internal/packager2/remove.go

```diff
@@ -0,0 +1,168 @@
+					if err != nil {
+						return err
+					}
+					client := action.NewUninstall(actionConfig)
+					client.KeepHistory = false
+					client.Wait = true
+					client.Timeout = config.ZarfDefaultTimeout
+					_, err = client.Run(chart.ChartName)
+					if err != nil && !errors.Is(err, driver.ErrReleaseNotFound) {
+						return fmt.Errorf("unable to uninstall the helm chart %s in the namespace %s: %w", chart.ChartName, chart.Namespace, err)
+					}
+					if errors.Is(err, driver.ErrReleaseNotFound) {
+						message.Warnf("Helm release for helm chart '%s' in the namespace '%s' was not found.  Was it already removed?", chart.ChartName, chart.Namespace)
+					}
+
+					// Pop the removed helm chart from the installed charts slice.
+					installedCharts := depPkg.DeployedComponents[len(depPkg.DeployedComponents)-1].InstalledCharts
+					installedCharts = installedCharts[:len(installedCharts)-1]
+					depPkg.DeployedComponents[len(depPkg.DeployedComponents)-1].InstalledCharts = installedCharts
+					err = opt.Cluster.UpdateDeployedPackage(ctx, *depPkg)
+					if err != nil {
+						// We warn and ignore errors because we may have removed the cluster that this package was inside of
+						message.Warnf("Unable to update the secret for package %s, this may be normal if the cluster was removed: %s", depPkg.Name, err.Error())
+					}
+				}
+			}
+
+			err = actions.Run(ctx, comp.Actions.OnRemove.Defaults, comp.Actions.OnRemove.After, nil)
+			if err != nil {
+				return fmt.Errorf("unable to run the after action: %w", err)
+			}
+			err = actions.Run(ctx, comp.Actions.OnRemove.Defaults, comp.Actions.OnRemove.OnSuccess, nil)
+			if err != nil {
+				return fmt.Errorf("unable to run the success action: %w", err)
+			}
+
+			// Pop the removed component from deploy components slice.
+			if opt.Cluster != nil {
+				depPkg.DeployedComponents = depPkg.DeployedComponents[:len(depPkg.DeployedComponents)-1]
+				err = opt.Cluster.UpdateDeployedPackage(ctx, *depPkg)
+				if err != nil {
+					// We warn and ignore errors because we may have removed the cluster that this package was inside of
+					message.Warnf("Unable to update the secret for package %s, this may be normal if the cluster was removed: %s", depPkg.Name, err.Error())
+				}
+			}
+			return nil
+		}()
+		if err != nil {
+			removeErr := actions.Run(ctx, comp.Actions.OnRemove.Defaults, comp.Actions.OnRemove.OnFailure, nil)
+			if removeErr != nil {
+				return errors.Join(fmt.Errorf("unable to run the failure action: %w", err), removeErr)
+			}
+			return err
+		}
+	}
+
+	// All the installed components were deleted, therefore this package is no longer actually deployed
+	if opt.Cluster != nil && len(depPkg.DeployedComponents) == 0 {
+		err := opt.Cluster.DeleteDeployedPackage(ctx, depPkg.Name)
+		if err != nil {
+			message.Warnf("Unable to delete the secret for package %s, tis may be normal if the cluster was removed: %s", depPkg.Name, err.Error())
```

Comment:
Sadly we cant because if the component we are deleting is actually deleting the cluster, we have removed the cluster we want to remove the secret from. So goes the life of chicken and the egg.